internal/adapters/rest/handlers/menu_handler.go

```diff
@@ -0,0 +1,190 @@
+package handlers
+
+import (
+	"github.com/B1gdawg0/se-project-backend/internal/infrastructure/entities"
+	"github.com/B1gdawg0/se-project-backend/internal/transaction/requests"
+	"github.com/B1gdawg0/se-project-backend/internal/usecases/menu"
+	"github.com/B1gdawg0/se-project-backend/internal/utils"
+	"github.com/gofiber/fiber/v2"
+)
+
+type MenuRestHandler struct {
+	usecase menu.MenuUseCase
+}
+
+func ProvideMenuRestHandler(usecase menu.MenuUseCase) *MenuRestHandler {
+	return &MenuRestHandler{
+		usecase: usecase,
+	}
+}
+
+func (mrh *MenuRestHandler) GetMenuByID(c *fiber.Ctx) error {
+	id := c.Params("id")
+
+	if id == "" {
+		return c.Status(fiber.StatusBadRequest).JSON(fiber.Map{
+			"message": "Bad Request",
+			"error":   "Must pass id to get the menu",
+		})
+	}
+
+	// Convert the string ID to a pgtype.UUID using utils.StringToUUID
+	_, err := utils.StringToUUID(id)
+	if err != nil {
+		return c.Status(fiber.StatusBadRequest).JSON(fiber.Map{
+			"message": "Bad Request",
+			"error":   "Invalid UUID format",
+		})
+	}
```

Comment:
These lines look like you want to check if it's UUID . Use utils.CheckUUID instead of utils.StringToUUID because StringToUUID create more overhead also wrong purpose